src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/AliasManager.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Messaging;
+using Microsoft.CmdPal.UI.ViewModels.Messages;
+
+namespace Microsoft.CmdPal.UI.ViewModels;
+
+public partial class AliasManager : ObservableObject
+{
+    private readonly TopLevelCommandManager _tlcManager;
+    private readonly Dictionary<string, CommandAlias> _aliases = [];
+
+    public AliasManager(TopLevelCommandManager tlcManager)
+    {
+        _tlcManager = tlcManager;
+        PopulateDefaultAliases();
+    }
+
+    private void AddAlias(CommandAlias a) => _aliases.Add(a.SearchPrefix, a);
+
+    public bool CheckAlias(string searchText)
+    {
+        if (_aliases.TryGetValue(searchText, out var alias))
+        {
+            try
+            {
+                var tlc = _tlcManager.LookupCommand(alias.CommandId);
+                if (tlc != null)
+                {
+                    WeakReferenceMessenger.Default.Send<ClearSearchMessage>();
+                    WeakReferenceMessenger.Default.Send<PerformCommandMessage>(new(new(tlc.Command)));
+                    return true;
+                }
+            }
+            catch
+            {
+            }
+        }
+
+        return false;
+    }
+
+    private void PopulateDefaultAliases()
+    {
+        this.AddAlias(new CommandAlias(":", "com.microsoft.cmdpal.registry", true));
+        this.AddAlias(new CommandAlias("$", "com.microsoft.cmdpal.windowsSettings", true));
+        this.AddAlias(new CommandAlias("=", "com.microsoft.cmdpal.calculator", true));
+        this.AddAlias(new CommandAlias(">", "com.microsoft.cmdpal.shell", true));
+        this.AddAlias(new CommandAlias("<", "com.microsoft.cmdpal.windowwalker", true));
```

Comment:
General thoughts I've been having regarding aliases that I'm not sure how to reason about just yet. This very well may be NOT the correct thread: 

1. Right now extensions have to input "Id" into their command for us to be able to identify that command - seems simple enough, except what happens if there is an Id collision? Example -- two extensions choose the exact same Id `ohibetthisisaquniqueid45663636363`. That is a really bad example, but maybe `id1`. 

2. Because of the req. that extensions manually bake in their own Id, only extensions that follow this convention would allow their commands to have aliases. As we think about the general settings situation -- this is going to get complicated because it's not just aliases anymore but also direct activation and keyboard shortcuts (in the long run). That means if an extension doesn't create an Id for a command that the user won't be able to benefit from things like creating aliases, but it feels like it should just work for all commands as a user.

3. It would also be great to be able to have aliases for applications themselves. We have that in a few mocks... maybe those would already have some unique ids though we could leverage.

Reply:
Don't overindex on the IDs. Terminal has a similar concept with the IDs for our actions, and if one isn't provided, we can always generate a secret one behind the scenes. We could use the PFN of the app the command came from, hash the title/subtitle/name, slap those together and presto - unique strings for almost anything. 

Self-assigning id's might make sense for an app developer who wants to actually manually invoke the palette (e.g. via a URI handler). Or, the spec also calls out affordances for apps to more efficiently invoke commands when we add support for stub commands, if they give us an ID. 

Picking relatively unique "namespace" strings for things like this is usually not very much of a problem. Chances are if you think your namespace will conflict with someone, it probably will, so pick something more unique. We can also manually reject IDs with the prefix `com.microsoft.cmdpal`

basically there's a lot more we can do here. This is just the beginning.

Reply:
That's #92 😉  I really gotta stand up an actual settings model so that we can build a settings viewmodel and settings page.

Reply:
oh man I really need to force myself to open Civilization and actually start my vacation before I just go do the whole settings model